src/agent/src/image_rpc.rs

```diff
@@ -44,6 +76,197 @@ impl ImageService {
         oci_path
     }
 
+    fn handle_layer_data(
+        image_data: &ImageData,
+        image_manifest: &mut OciManifest,
+        crypto_config: &CryptoConfig,
+        oci_blob_path: &str,
+    ) -> Result<()> {
+        for layer in image_data.layers.iter() {
+            let layer_digest = layer.clone().sha256_digest();
+
+            if layer.media_type == MEDIA_TYPE_LAYEE_GZIP_ENC
+                || layer.media_type == MEDIA_TYPE_LAYER_ENC
+            {
+                if crypto_config.decrypt_config.is_none() {
```

Comment:
Can we use the

```rust
if let Some(ref decrypt_config: &{unknown}) = crypto_config.decrypt_config {
}
```

pattern instead, and avoid unwrapping later?

Reply:
Yes, thanks, fixed now.